lib/parser.dart

```diff
@@ -2209,10 +2217,11 @@ class _Parser {
   }
 
   // TODO(terry): Need to handle auto.
-  marginValue(var exprTerm) {
+  num marginValue(var exprTerm) {
     if (exprTerm is UnitTerm || exprTerm is NumberTerm) {
-      return exprTerm.value;
+      return exprTerm.value as num;
```

Comment:
If the do make `exprTerm` a `LiteralTerm`, the `value` field will cease to be accessible here (it's currently dynamic dispatch). I suggest separate conditional clauses for each term.

Reply:
Good catch, done.